### PR TITLE
Use docker for development

### PR DESCRIPTION
**Pull request by hermzz**
Creates two docker environments: `api` and `node`.
Setups up CORS and removes the proxy to the API through React.
With CORS we can make requests to other domains/ports.

### CODE REVIEW

File: frontend/modules/common/config.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
+var apiHost = "http://localhost:8000"
+var apiUrl = apiHost + '/api/v1'
 
 export var serverURLs = {
-  auth_token: '/api/v1/accounts/obtain-auth-token/',
-  browse: '/api/v1/recipe/recipes/?format=json&fields=id,title,pub_date,rating,photo_thumbnail,info',
-  mini_browse: '/api/v1/recipe/mini-browse/?format=json',
```
Comment (RyanNoelk) [inline frontend/modules/common/config.js:5]: This endpoint is needed. Its used to return a small (or large) random subset or recipes to be used on the home page and when viewing individual recipes.

File: frontend/modules/common/config.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
+var apiHost = "http://localhost:8000"
+var apiUrl = apiHost + '/api/v1'
```
Comment (RyanNoelk) [inline frontend/modules/common/config.js]: semicolons ^^

File: frontend/package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start": "if-env NODE_ENV=production && npm run start:prod || npm run start:dev",
     "start:dev": "webpack-dev-server --inline --content-base public --history-api-fallback",
-    "start:prod": "webpack && node server.js",
+    "start:prod": "NODE_ENV=production webpack && node server.js",
```
Comment (RyanNoelk) [inline frontend/package.json]: This command shouldn't be called directly. 

running `NODE_ENV=production npm start` will call `start:prod` and run them with `NODE_ENV=production`

**Review by RyanNoelk — CHANGES_REQUESTED** [review]
Thanks a ton for adding some much needed docker support! A few questions/comments: 

What are your thoughts on dockerizing the DB as well? I don't see a good reason for not doing it at this stage. 

I use OSX for development and I'm having some issues with the volume mounting for the frontend. I'm thinking that this is just a file watching issue, since the django server will auto update if I change a file. I'll take a closer look into it this weekend, if things are working for you.

Thanks!
-Ryan

**Comment by hermzz** [comment]
Dockerizing the DB should be fairly straight forward. I hadn't done it because the sqlite DB seemed to be fine, but I can give it a shot.

As for the volume issues, I seem to recall reading about people having trouble with it because of boot2docker. I think there was a quick fix that just required manually setting a `tick` setting. I'll try to find it.

File: frontend/modules/common/config.js (hunk 1 above)
Comment (hermzz) [reply frontend/modules/common/config.js:5]: Added back, I think it got lost when I was merging changes.

**Comment by RyanNoelk** [comment]
l'm mainly asking about the DB because I would like to use this for production soon as well. Downloading docker images is a lot more straight forward than building the code from source

File: frontend/webpack.config.js [hunk 4]
```diff
@@ -29,28 +51,15 @@ module.exports = {
...
-        secure: false
-      }
-    }
+    inline: true,
+    host: "0.0.0.0",
+    port: "8080"
+  },
+
+  watchOptions: {
+    poll: 100,
```
Comment (hermzz) [inline frontend/webpack.config.js]: @RyanNoelk Play around with these values. Apparently this is what fixes the refreshing issue on Mac.

**Comment by RyanNoelk** [comment]
FYI, I did some quite digging and found a couple possible solutions to my mac problems. 

* docker-osx-dev - https://github.com/brikis98/docker-osx-dev
* Running docker natively - https://www.docker.com/products/overview - Looks like docker has a OSX beta.  

I'll also give the above a try as well!

**Comment by RyanNoelk** [comment]
So the issue went away once I switched to native docker for mac. Having some cors related issues that I'll take a look at this weekend.

File: frontend/webpack.config.js (hunk 4 above)
Comment (RyanNoelk) [reply frontend/webpack.config.js]: Feel free to remove this. It's not needed anymore.

**Review by RyanNoelk — APPROVED** [review]

**Comment by RyanNoelk** [comment]
LGTM!

**Pull request merged**